From 2dcfd52287618fb29882aaf9e514c72a9593a8d3 Mon Sep 17 00:00:00 2001
From: Maciej Borzecki <maciej.borzecki@rndity.com>
Date: Wed, 9 Aug 2017 11:13:12 +0200
Subject: [PATCH] colibri-imx7: set CONFIG_ENV_RANGE to cover 2 * environment
 size

Signed-off-by: Maciej Borzecki <maciej.borzecki@rndity.com>
---
 include/configs/colibri_imx7.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/include/configs/colibri_imx7.h b/include/configs/colibri_imx7.h
index 9eb4bfa..da69ccb 100644
--- a/include/configs/colibri_imx7.h
+++ b/include/configs/colibri_imx7.h
@@ -202,6 +202,7 @@
 /* configured by mender integration */
 /* #define CONFIG_ENV_OFFSET		(28 * CONFIG_ENV_SECT_SIZE) */
 #define CONFIG_ENV_SIZE			CONFIG_ENV_SECT_SIZE
+#define CONFIG_ENV_RANGE    (2 * CONFIG_ENV_SIZE)
 #else
 #define CONFIG_ENV_SIZE			(8 * SZ_64K)
 #endif
-- 
2.7.4

