From 3bea2ead650dab2d2fa55f79f4865aeacd8ad4dd Mon Sep 17 00:00:00 2001
From: Drew Moseley <drew.moseley@northern.tech>
Date: Wed, 22 Nov 2017 15:40:03 -0500
Subject: [PATCH] colibri-imx7: mender integration, override ubiboot and
 bootcmd

Signed-off-by: Maciej Borzecki <maciej.borzecki@rndity.com>
Signed-off-by: Drew Moseley <drew.moseley@northern.tech>
---
 include/configs/colibri_imx7.h | 17 ++++++++++-------
 1 file changed, 10 insertions(+), 7 deletions(-)

diff --git a/include/configs/colibri_imx7.h b/include/configs/colibri_imx7.h
index da69ccb..49e2707 100644
--- a/include/configs/colibri_imx7.h
+++ b/include/configs/colibri_imx7.h
@@ -104,22 +104,25 @@
 		"bootm ${ramdisk_addr_r}#config@${soc}\0"
 #else /* CONFIG_TDX_EASY_INSTALLER */
 #define UBI_BOOTCMD \
-	"ubiargs=ubi.mtd=ubi root=ubi0:rootfs rw rootfstype=ubifs " \
+	"ubiargs=ubi.mtd=${mender_mtd_ubi_dev_name} root=${mender_kernel_root} rw rootfstype=ubifs " \
 		"ubi.fm_autoconvert=1\0" \
 	"ubiboot=run setup; " \
-		"setenv bootargs ${defargs} ${ubiargs} " \
+		"setenv fdtfile ${soc}-colibri-${fdt_board}.dtb; "  \
+		"setenv bootargs ${defargs} ${ubiargs} ${mtdparts} " \
 		"${setupargs} ${vidargs}; echo Booting from NAND...; " \
-		"ubi part ubi && run m4boot && " \
-		"ubi read ${kernel_addr_r} kernel && " \
-		"ubi read ${fdt_addr_r} dtb && " \
-		"run fdt_fixup && bootz ${kernel_addr_r} - ${fdt_addr_r}\0"
+		"run mender_setup; "                                   \
+		"ubi part ${mender_mtd_ubi_dev_name} && "              \
+		"ubifsmount ${mender_uboot_root_name} && "             \
+		"ubifsload ${kernel_addr_r} /boot/zImage && "          \
+		"ubifsload ${fdt_addr_r} /boot/${fdtfile} && "         \
+		"bootz ${kernel_addr_r} - ${fdt_addr_r}\0"
 #endif /* CONFIG_TDX_EASY_INSTALLER */
 
 #ifdef CONFIG_TDX_EASY_INSTALLER
 #define CONFIG_BOOTCOMMAND "setenv fdtfile ${soc}-colibri-${fdt_board}.dtb && " \
 	"run bootcmd_mmc0; run ubiboot; run distro_bootcmd"
 #else
-#define CONFIG_BOOTCOMMAND "run ubiboot; " \
+#define CONFIG_BOOTCOMMAND "run ubiboot; run mender_try_to_recover; " \
 	"setenv fdtfile ${soc}-colibri-${fdt_board}.dtb && run distro_bootcmd"
 #endif
 
-- 
2.7.4

