From a16097c709a93d6c0502a5932c1f8164608a4f24 Mon Sep 17 00:00:00 2001
From: Kristian Amlie <kristian.amlie@mender.io>
Date: Mon, 15 Feb 2016 15:57:32 +0100
Subject: [PATCH] Add missing header which fails on recent GCC.

---
 tools/env/fw_env.c |    1 +
 1 file changed, 1 insertion(+)

diff --git a/tools/env/fw_env.c b/tools/env/fw_env.c
index daa02a7..7a194b5 100644
--- a/tools/env/fw_env.c
+++ b/tools/env/fw_env.c
@@ -16,6 +16,7 @@
 #include <linux/stringify.h>
 #include <stdio.h>
 #include <stdlib.h>
+#include <stdint.h>
 #include <stddef.h>
 #include <string.h>
 #include <sys/types.h>
-- 
1.7.9.5

