From 2758cc7c0d04db3035172ad0f7eaa024055f83e6 Mon Sep 17 00:00:00 2001
Message-Id: <2758cc7c0d04db3035172ad0f7eaa024055f83e6.1502452850.git.maciej.borzecki@rndity.com>
From: Maciej Borzecki <maciej.borzecki@rndity.com>
Date: Fri, 11 Aug 2017 14:00:09 +0200
Subject: [PATCH] colibri-imx7: Mender QA - reset the board on panic

Signed-off-by: Maciej Borzecki <maciej.borzecki@rndity.com>
---
 include/configs/colibri_imx7.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/configs/colibri_imx7.h b/include/configs/colibri_imx7.h
index d3dd21b5c1d53e58d35e7fe75a7a066aba8c16b9..e8379a9e72aef74ff3d0ea5279bd27b4fd5658c3 100644
--- a/include/configs/colibri_imx7.h
+++ b/include/configs/colibri_imx7.h
@@ -138,7 +138,7 @@
 	SD_BOOTCMD \
 	UBI_BOOTCMD \
 	"console=ttymxc0\0" \
-	"defargs=\0" \
+	"defargs=panic=5\0" \
 	"fdt_board=eval-v3\0" \
 	"fdt_fixup=;\0" \
 	"m4boot=;\0" \
-- 
2.9.4

