From ff3328d0c1eca266aac9fbebd0ce7317589cbb99 Mon Sep 17 00:00:00 2001
Message-Id: <ff3328d0c1eca266aac9fbebd0ce7317589cbb99.1502270128.git.maciej.borzecki@rndity.com>
From: Maciej Borzecki <maciej.borzecki@rndity.com>
Date: Tue, 1 Aug 2017 08:11:53 +0200
Subject: [PATCH] colibri-imx7: mender integration, override ubiboot and
 bootcmd

Signed-off-by: Maciej Borzecki <maciej.borzecki@rndity.com>

Signed-off-by: Maciej Borzecki <maciej.borzecki@rndity.com>
---
 include/configs/colibri_imx7.h | 21 ++++++++++++---------
 1 file changed, 12 insertions(+), 9 deletions(-)

diff --git a/include/configs/colibri_imx7.h b/include/configs/colibri_imx7.h
index d06d74bcfaaf0630f7f012ae3edf1de9109a6e67..d09a514bb80f2a43f342766ef19cd298c0c37cee 100644
--- a/include/configs/colibri_imx7.h
+++ b/include/configs/colibri_imx7.h
@@ -94,17 +94,20 @@
 		"run fdt_fixup && bootz ${kernel_addr_r} - ${fdt_addr_r}\0" \
 
 #define UBI_BOOTCMD	\
-	"ubiargs=ubi.mtd=ubi root=ubi0:rootfs rootfstype=ubifs " \
+	"ubiargs=ubi.mtd=${mender_mtd_ubi_dev_name} root=${mender_kernel_root} rootfstype=ubifs " \
 		"ubi.fm_autoconvert=1\0" \
-	"ubiboot=run setup; " \
-		"setenv bootargs ${defargs} ${ubiargs} " \
+		"ubiboot=run setup; " \
+		"setenv fdtfile ${soc}-colibri-${fdt_board}.dtb; "  \
+		"setenv bootargs ${defargs} ${ubiargs} ${mtdparts} " \
 		"${setupargs} ${vidargs}; echo Booting from NAND...; " \
-		"ubi part ubi && run m4boot && " \
-		"ubi read ${kernel_addr_r} kernel && " \
-		"ubi read ${fdt_addr_r} dtb && " \
-		"run fdt_fixup && bootz ${kernel_addr_r} - ${fdt_addr_r}\0" \
-
-#define CONFIG_BOOTCOMMAND "run ubiboot; " \
+		"run mender_setup; "                                   \
+		"ubi part ${mender_mtd_ubi_dev_name} && "              \
+  	"ubifsmount ${mender_uboot_root_name} && "             \
+  	"ubifsload ${kernel_addr_r} /boot/zImage && "          \
+  	"ubifsload ${fdt_addr_r} /boot/${fdtfile} && "         \
+  	"bootz ${kernel_addr_r} - ${fdt_addr_r}\0"             \
+
+#define CONFIG_BOOTCOMMAND "run ubiboot; run mender_try_to_recover; " \
 	"setenv fdtfile ${soc}-colibri-${fdt_board}.dtb && run distro_bootcmd;"
 
 #define BOOTENV_RUN_NET_USB_START ""
-- 
2.9.4

