From 3345c9a48aae4769ceafce77306e925dba91cdcd Mon Sep 17 00:00:00 2001
From: Kristian Amlie <kristian.amlie@northern.tech>
Date: Fri, 16 Feb 2018 18:15:28 +0100
Subject: [PATCH 4/4] Remove MTDIDS and MTDPARTS defines. Will be defined by
 Mender.

Signed-off-by: Kristian Amlie <kristian.amlie@northern.tech>
---
 include/configs/colibri_imx7.h | 7 -------
 1 file changed, 7 deletions(-)

diff --git a/include/configs/colibri_imx7.h b/include/configs/colibri_imx7.h
index ea04762..53aa35d 100644
--- a/include/configs/colibri_imx7.h
+++ b/include/configs/colibri_imx7.h
@@ -204,13 +204,6 @@
 #define CONFIG_CMD_MTDPARTS	/* Enable 'mtdparts' command line support */
 #define CONFIG_MTD_PARTITIONS
 #define CONFIG_MTD_DEVICE	/* needed for mtdparts commands */
-#define MTDIDS_DEFAULT		"nand0=gpmi-nand"
-#define MTDPARTS_DEFAULT	"mtdparts=gpmi-nand:"		\
-				"512k(mx7-bcb),"		\
-				"1536k(u-boot1)ro,"		\
-				"1536k(u-boot2)ro,"		\
-				"512k(u-boot-env),"		\
-				"-(ubi)"
 
 /* DMA stuff, needed for GPMI/MXS NAND support */
 #define CONFIG_APBH_DMA
-- 
2.7.4

