From 8b0b8f362964f42559b39488d0c529b6e2b99523 Mon Sep 17 00:00:00 2001
From: Kristian Amlie <kristian.amlie@mender.io>
Date: Mon, 15 Feb 2016 15:57:32 +0100
Subject: [PATCH] Add missing header which fails on recent GCC.

---
 tools/env/fw_env.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/tools/env/fw_env.c b/tools/env/fw_env.c
index 3dc0d53..3aac340 100644
--- a/tools/env/fw_env.c
+++ b/tools/env/fw_env.c
@@ -19,6 +19,7 @@
 #include <ctype.h>
 #include <stdio.h>
 #include <stdlib.h>
+#include <stdint.h>
 #include <stddef.h>
 #include <string.h>
 #include <sys/types.h>
-- 
2.7.4

