From 3f906dda522acc2aba18ec57e831cfcdfff9d4a2 Mon Sep 17 00:00:00 2001
From: Kristian Amlie <kristian.amlie@northern.tech>
Date: Thu, 15 Mar 2018 17:15:21 +0100
Subject: [PATCH 4/4] Set UBI erase size from Mender provided define.

Signed-off-by: Kristian Amlie <kristian.amlie@northern.tech>
---
 tools/env/fw_env.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/tools/env/fw_env.c b/tools/env/fw_env.c
index 18c2324..7d6228a 100644
--- a/tools/env/fw_env.c
+++ b/tools/env/fw_env.c
@@ -1539,6 +1539,15 @@ static int check_device_config(int dev)
 				DEVNAME(dev));
 			goto err;
 		}
+#ifdef MENDER_UBI_LEB_SIZE
+		if (DEVESIZE(dev) == 0) {
+			/*
+			 * It would be better if this was detected at run time,
+			 * but this is the next best thing.
+			 */
+			DEVESIZE(dev) = MENDER_UBI_LEB_SIZE;
+		}
+#endif
 	} else if (S_ISCHR(st.st_mode)) {
 		struct mtd_info_user mtdinfo;
 		rc = ioctl(fd, MEMGETINFO, &mtdinfo);
-- 
2.7.4

